### ISSUE TITLE
Qualified record update

### ISSUE THREAD

**Issue by philderbeast** [elm-lang/elm-compiler#1321]
I can reference a record declared in a module using a qualified name. I can update from that record if it is locally bound. I cannot update from the qualified name of the record.

The reproduction of this is in the repo at https://github.com/philderbeast/elm-record-qualified

```
-- A.elm
module A (B, zero) where

type alias B =
    { ii : Int }

zero : B
zero =
    { ii = 0 }

-- Main.elm
import A as A exposing (zero)

unqualified =
    let local =
            A.zero

        localUpdate =
            { local | ii = 1 }

qualified =
    let qualifiedUpdate =
            { A.zero | ii = 3 } -- this fails to compile
```

```
E:\Dev\Src\elm-record-qualified>elm-make --warn src\Main.elm
-- SYNTAX PROBLEM ------------------------------------------------- src\Main.elm

I ran into something unexpected when parsing your code!

33│             { A.zero | ii = 1 }
                  ^
I am looking for one of the following things:

    a closing bracket '}'
    a lower case name
    whitespace

Detected errors in 1 module.
```

**Comment by jvoigtlaender** [comment]
This is a duplicate. Please clowe.

**Comment by philderbeast** [comment]
Do you have the duplicate issue number?

**Comment by jvoigtlaender** [comment]
Yes, it's covered by #1068.

**Comment by philderbeast** [comment]
OK, yes this is much the same as #635 that is collected on #1068 with #635 itself being closed in favour of elm-lang/elm-plans#16. This case #1321 uses import X as Y and then Y.record whereas #635 uses the import X and then X.record. Both fail on 0.16 and both are deserving as test cases if we were to support qualified names in record update.
